torch/autograd/_functions/tensor.py

```diff
@@ -191,7 +190,7 @@ def backward(ctx, grad_output):
 
         if ctx.needs_input_grad[3]:
             index, = ctx.saved_tensors
-            grad_tensor2 = grad_output.index_select(ctx.dim, index)
+            grad_tensor2 = grad_output.index_select(ctx.dim, Variable(index))
```

Comment:
You should use `saved_variables` instead of repacking index (same for All other index functions)